src/components/DocumentationTopic.vue

```diff
@@ -732,6 +732,12 @@ export default {
 
     .source {
       border-radius: var(--code-border-radius);
+
+      > code {
+        @include inTargetIde() {
+          white-space: pre-wrap;
```

Comment:
I think I've already expressed my opinion on this before, but I generally don't think it's a great idea to try and mix our current hard wrapping approach with a soft wrapping approach for displaying symbol declarations.

Just as a quick example, here is a comparison of how an example Swift API with multiple params would look with the current approach vs the `pre-wrap` here:

(current)
<img width="1436" alt="Screenshot 2023-05-08 at 2 05 02 PM" src="https://user-images.githubusercontent.com/212918/236935400-1a0d23ed-5c5f-4190-9a25-633fe7661f4a.png">

(`white-space: pre-wrap`)
<img width="1436" alt="Screenshot 2023-05-08 at 2 04 54 PM" src="https://user-images.githubusercontent.com/212918/236935421-ba146ab7-efb5-46f2-acde-59dba3b3dbe9.png">

This is just my subjective opinion, but I think we should go with either hard-wrap or soft-wrap and not try to mix—and my preference would be to keep the hard-wrapping since it breaks up parameters in a clearer way.

Reply:
Unless this is a user specific option, aka per IDE, I dont like it either. If a user prefers wrapping, then we could add it for them, but to all by default I dont think its great.

Another thing, wrapping should probably keep the indentation it's line, which i am not sure how to do...

Reply:
FWIW, this PR only allows the declaration to wrap when it's in both IDE and minimized mode. To Ethan's point, this is a really specific use case where we don't want the users to have to use the mouse to see the full content of the declaration.

Reply:
> > Another thing, wrapping should probably keep the indentation it's line, which i am not sure how to do...
> 
> Any ideas on how to achieve this? This is my main feedback as well. From discussions with Hanqing it seems like that would be a much larger effort.

Assuming that you mean a single line by "it's line", we would probably need to update the [logic](https://github.com/apple/swift-docc-render/blob/864361eb64e16de58e4257fe9b86fb8e70d96156/src/components/DocumentationTopic/PrimaryContent/DeclarationSource.vue#L60) to _not_ use the formatted tokens and instead use the original tokens in this specific scenario—that way we would just be pivoting to soft wrapping and not mixing the soft wrapping with the hard wrapping we do today. I still think it would be a worse experience though, but that's just my subjective opinion.